.github/workflows/ci.yml

```diff
@@ -0,0 +1,87 @@
+name: ci
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    outputs:
+      branch: ${{ steps.check_step.outputs.branch }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Get current branch
+        id: check_step
+        run: |
+          if ${{ github.event_name == 'pull_request' }}
+          then
+            branch=${{ github.base_ref }}
+          else
+            raw=$(git branch -r --contains ${{ github.ref }})
+            branch=${raw##*/}
+          fi
+          echo "branch=$branch" >> $GITHUB_OUTPUT
+          echo "Branch is $branch."
+
+  build:
+    needs: check # Wait for check step to finish
+    runs-on: ubuntu-latest
+    # minimal permissions for Get Artifact Publish Token
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19.7
```

Comment:
We have a couple places where the go version is specified, they should be updated to refer to this, and this should be updated to refer to them, specifically:
- readme
- makefile
- dockerfiles

Reply:
Okay... Will comment